src/scanspec/specs.py

```diff
@@ -108,7 +120,256 @@ def serialize(self) -> Mapping[str, Any]:
+        assert set(left_axes) == set(right_axes), f"axes {left_axes} != {right_axes}"
+        return left_axes
+
+    def calculate(self, bounds=True, nested=False) -> List[Frames[Axis]]:
+        dim_left = squash_frames(
+            self.left.calculate(bounds, nested), nested and self.check_path_changes
+        )
+        dim_right = squash_frames(
+            self.right.calculate(bounds, nested), nested and self.check_path_changes
+        )
+        dim = dim_left.concat(dim_right, self.gap)
+        return [dim]
+
+
+@dataclass(config=StrictConfig)
+class Line(Spec[Axis]):
+    """Linearly spaced frames with start and stop as first and last midpoints.
+
+    .. example_spec::
+
+        from scanspec.specs import Line
+
+        spec = Line("x", 1, 2, 5)
+    """
+
+    axis: Axis = Field(description="An identifier for what to move")
+    start: float = Field(description="Midpoint of the first point of the line")
+    stop: float = Field(description="Midpoint of the last point of the line")
+    num: int = Field(ge=1, description="Number of frames to produce")
+
+    def axes(self) -> List:
+        return [self.axis]
+
+    def _line_from_indexes(self, indexes: np.ndarray) -> Dict[Axis, np.ndarray]:
+        if self.num == 1:
+            # Only one point, stop-start gives length of one point
+            step = self.stop - self.start
+        else:
+            # Multiple points, stop-start gives length of num-1 points
+            step = (self.stop - self.start) / (self.num - 1)
+        # self.start is the first centre point, but we need the lower bound
+        # of the first point as this is where the index array starts
+        first = self.start - step / 2
+        return {self.axis: indexes * step + first}
+
+    def calculate(self, bounds=True, nested=False) -> List[Frames[Axis]]:
+        return _dimensions_from_indexes(
+            self._line_from_indexes, self.axes(), self.num, bounds
+        )
+
+    @classmethod
+    def bounded(
+        cls,
+        axis: Axis = Field(description="An identifier for what to move"),
+        lower: float = Field(description="Lower bound of the first point of the line"),
+        upper: float = Field(description="Upper bound of the last point of the line"),
+        num: int = Field(ge=1, description="Number of frames to produce"),
+    ) -> Line[Axis]:
+        """Specify a Line by extreme bounds instead of midpoints.
+
+        .. example_spec::
+
+            from scanspec.specs import Line
+
+            spec = Line.bounded("x", 1, 2, 5)
+        """
+        half_step = (upper - lower) / num / 2
+        start = lower + half_step
+        if num == 1:
+            # One point, stop will only be used for step size
+            stop = upper + half_step
+        else:
+            # Many points, stop will be produced
+            stop = upper - half_step
+        return cls(axis, start, stop, num)
+
+
+Line.bounded = validate_call(Line.bounded)
```

Comment:
Document why this is necessary (field validation for class methods that return a forward referenced: forward reference not available when the decorator would be processed)